### PR TITLE
[ML] address potential bug where trained models get stuck in starting after being allocated to node

### PR DESCRIPTION
**Pull request by benwtrent**
When a model is starting, it has been rarely observed that it will lock up while trying to restore the model objects to the native process. 

This would manifest as a trained model being stuck in "starting" while also being assigned to a node. So, there is a native process started and task available on the assigned nodes, but the model state never gets out of "starting".

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java [hunk 1]
```diff
@@ -149,34 +150,46 @@ private void doStartDeployment(TrainedModelDeploymentTask task, ActionListener<T
...
-                // `startAndLoad` creates named pipes, blocking the calling thread, better to execute that in our utility
-                // executor.
-                executorServiceForDeployment.execute(() -> startAndLoad(processContext, modelConfig.getLocation(), modelLoadedListener));
-            }, listener::onFailure));
+                }, listener::onFailure));
+            } else {
+                listener.onFailure(
+                    new IllegalArgumentException(
+                        format(
+                            "[%s] must be an pytorch model found inference config of kind [%s]",
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java:186]:
```suggestion
                            "[%s] must be a pytorch model; found inference config of kind [%s]",
```

**Review by dimitris-athanasiou — COMMENTED** [review]
LGTM Just a suggestion for an error message.

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by dimitris-athanasiou — APPROVED** [review]

**Pull request merged**